src/main/java/com/bazel_diff/BazelSourceFileTarget.java

```diff
@@ -20,24 +20,24 @@
 
     BazelSourceFileTargetImpl(String name, byte[] digest, Path workingDirectory) throws IOException, NoSuchAlgorithmException {
         this.name = name;
-        byte[] data = null;
+        MessageDigest finalDigest = MessageDigest.getInstance("SHA-256");
         if (workingDirectory != null && name.startsWith("//")) {
             String filenameSubstring = name.substring(2);
             String filenamePath = filenameSubstring.replaceFirst(":", "/");
-            File sourceFile = new File(workingDirectory.toString(), filenamePath);
-            if (sourceFile.isFile() && sourceFile.canRead()) {
-                ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
-                outputStream.write(Files.readAllBytes(sourceFile.toPath()));
-                outputStream.write(digest);
-                data = outputStream.toByteArray();
-                outputStream.close();
-            }
-        } else {
-            data = digest;
-        }
-        MessageDigest finalDigest = MessageDigest.getInstance("SHA-256");
-        if (data != null) {
-            finalDigest.update(data);
+            Path absoluteFilePath = Paths.get(workingDirectory.toString(), filenamePath);
+            try (RandomAccessFile sourceFile = new RandomAccessFile(absoluteFilePath.toString(), "r")) {
+                FileChannel inChannel = sourceFile.getChannel();
+                long fileSize = inChannel.size();
+                ByteBuffer buffer = ByteBuffer.allocate((int) fileSize);
```

Comment:
You probably want to read in chunks, say 1024 bytes, instead of allocating a huge ByteByffer. Reason being that it potentially can lead to OOM and the underlying hashing is usually done via streaming chunks anyways

Reply:
I tried
```
        while (inChannel.read(buffer) != -1) {
            buffer.flip();
            finalDigest.update(buffer);
            buffer.clear();
        }
```
but that makes content hash significant slower (150s-> 200s).😂